### ISSUE TITLE
line 59, in _main: can't concat bytes to str

### ISSUE THREAD

**Issue by thorade** [nschloe/betterbib#29]
When I try to run the script, I get the following error:

```
Reading from: references.bib
Saving to: refout.bib
|----------| 0/368   0% [elapsed: 00:00 left: ?, ? iters/sec]Traceback (most recent call last):
  File "C:\Users\mthorade\Documents\Python\WinPython-64bit-3.4.2.3\python-3.4.2.amd64\Scripts\betterbib", line 99, in <module>
    _main()
  File "C:\Users\mthorade\Documents\Python\WinPython-64bit-3.4.2.3\python-3.4.2.amd64\Scripts\betterbib", line 59, in _main
    '\n\n'
TypeError: can't concat bytes to str
```

**Comment by nschloe** [comment]
Woaw, okay. Can you post a minimal `references.bib` that reproduces the error?

**Comment by thorade** [comment]
Looks like it will happen on any file, I used this one:
```
@article{Hopfe.2011,
  author = {Hopfe, Christina J. and Hensen, Jan L. M.},
  title = {Uncertainty analysis in building performance simulation for design support},
}
```

Probably I should run the tests that you deliver with your package.
My platform: WinPython-64bit-3.4.2.3 on Windows 10 64bit.

**Comment by thorade** [comment]
Running pytest gives following output:
```
C:\Users\mthorade\Documents\Python\betterbib>pytest
going into C:\Users\mthorade\Documents\Python\betterbib\test
=========================  test_crossref.py  =========================

no test dir found testing here: C:\Users\mthorade\Documents\Python\betterbib
*******************************************************************************
Ran 0 test cases in 0.08s (0.08s CPU)
All 1 modules OK
C:\Users\mthorade\Documents\Python\betterbib>cd test

C:\Users\mthorade\Documents\Python\betterbib\test>pytest
going into C:\Users\mthorade\Documents\Python\betterbib\test
=========================  test_crossref.py  =========================

no test dir found testing here: C:\Users\mthorade\Documents\Python\betterbib\test
=========================  test_crossref.py  =========================

*******************************************************************************
Ran 0 test cases in 0.08s (0.08s CPU)
All 2 modules OK
```

**Comment by nschloe** [comment]
Hm, I can't reproduce the error yet. For me, betterbib exits cleanly and gives
```
@article{Hopfe.2011,
  author = {Hopfe, Christina J. and Hensen, Jan L.M.},
  publisher = {Elsevier BV},
  doi = {10.1016/j.enbuild.2011.06.034},
  title = {Uncertainty analysis in building performance simulation for design support},
  url = {http://dx.doi.org/10.1016/j.enbuild.2011.06.034},
  journal = {Energy and Buildings},
  number = {10},
  month = oct,
  volume = {43},
  source = {CrossRef},
  year = {2011},
  pages = {2798-2805}
}
```

**Comment by thorade** [comment]
Yes, that would be perfect. So my installation seems to be the problem.

Here is another strange error:
The input 
```
@inproceedings{KraeuchiEtAl2014,
  title = {Modellbildung f\"ur thermische {Arealvernetzung} mit {IDA-ICE}},
  author = {Kr\"auchi, Philipp and Kolb, M. and Gautschi, Thomas and Menti, Urs-Peter and Sulzer, Matthias},
  booktitle = {BauSIM 2014},
  address = {Aachen, Germany},
  month = sep,
  url = {http://www.ibpsa.org/proceedings/bausimPapers/2014/p1130_final.pdf},
  year = {2014}
}
```
gives the error:
```
C:\Users\mthorade\Documents\git\iea-annex-60-admin\finalReport\source>python C:\Users\mthorade\Documents\Python\WinPython-64bit-3.4.2.3\python-3.4.2.amd64\Scripts\betterbib refin.bib refout.bib
Reading from: refin.bib
Saving to: refout.bib
|----------| 0/1   0% [elapsed: 00:00 left: ?, ? iters/sec]Traceback (most recent call last):
  File "C:\Users\mthorade\Documents\Python\WinPython-64bit-3.4.2.3\python-3.4.2.amd64\Scripts\betterbib", line 99, in <module>
    _main()
  File "C:\Users\mthorade\Documents\Python\WinPython-64bit-3.4.2.3\python-3.4.2.amd64\Scripts\betterbib", line 66, in _main
    % e.message
UnboundLocalError: local variable 'e' referenced before assignment
```

**Comment by thorade** [comment]
Just installed betterbib in WinPython 2.7.9 32bit and then the simple example works, same output as you showed.

The full references.bib ile with 300+ entries breaks with error:
```
C:\Users\mthorade\Documents\git\iea-annex-60-admin\finalReport\source>python C:\Users\mthorade\Documents\Python\WinPython-32bit-2.7.9.2\python-2.7.9\Scripts\betterbib references.bib refout.bib
Reading from: references.bib
Saving to: refout.bib

|----------| 9/365   2% [elapsed: 00:10 left: 07:12,  0.82 iters/sec]Traceback (most recent call last):
  File "C:\Users\mthorade\Documents\Python\WinPython-32bit-2.7.9.2\python-2.7.9\Scripts\betterbib", line 99, in <module>
    _main()
  File "C:\Users\mthorade\Documents\Python\WinPython-32bit-2.7.9.2\python-2.7.9\Scripts\betterbib", line 48, in _main
    result = source.find_unique(entry)
  File "C:\Users\mthorade\Documents\Python\WinPython-32bit-2.7.9.2\python-2.7.9\lib\site-packages\betterbib\crossref.py", line 160, in find_unique
    assert r.ok
AssertionError
```

**Comment by thorade** [comment]
This seems to be one of the problematic entries:
```
@InProceedings{KraeuchiEtAl2015:1,
  author = {Kr\"auchi, Philipp and Schluck, Thomas and Sulzer, Matthias},
  title = {Modelling of low temperature heating networks with IDA-ICE},
  booktitle = {Proceedings of International Conference CISBAT},
  doi = {10.5075/epfl-cisbat2015-827-832},
  pages = {827--832},
  address = {Lausanne, Switzerland},
  month = sep,
  year = {2015}
}
```
results in 
```
Reading from: refin.bib
Saving to: refout.bib

|----------| 0/1   0% [elapsed: 00:00 left: ?, ? iters/sec]Traceback (most recent call last):
  File "C:\Users\mthorade\Documents\Python\WinPython-32bit-2.7.9.2\python-2.7.9\Scripts\betterbib", line 99, in <module>
    _main()
  File "C:\Users\mthorade\Documents\Python\WinPython-32bit-2.7.9.2\python-2.7.9\Scripts\betterbib", line 48, in _main
    result = source.find_unique(entry)
  File "C:\Users\mthorade\Documents\Python\WinPython-32bit-2.7.9.2\python-2.7.9\lib\site-packages\betterbib\crossref.py", line 189, in find_unique
    if result['page'] == d['pages']:
KeyError: 'page'
```

If I delete the line ` pages = {827--832},` the error disappears.

**Comment by nschloe** [comment]
Thanks! I was able to reproduce it with Python 3.*. Fixing it now...

**Comment by thorade** [comment]
Thanks for all the quick answers & fixing it!

**Comment by nschloe** [comment]
Fixed in 2.1.1.

**Comment by thorade** [comment]
Yes, works for me!
